internal/hcl/evaluator.go

```diff
@@ -593,16 +593,26 @@ func expandCountBlockToValue(b *Block, existingValues map[string]cty.Value) cty.
 	return cty.TupleVal(elements)
 }
 
-func expandedEachBlockToValue(b *Block, existingValues map[string]cty.Value) cty.Value {
+func (e *Evaluator) expandedEachBlockToValue(b *Block, existingValues map[string]cty.Value) cty.Value {
 	k := b.Key()
 	if k == nil {
 		return cty.NilVal
 	}
 
 	ob := make(map[string]cty.Value)
 
-	eachMap := existingValues[stripCount(b.Labels()[1])]
+	name := b.Labels()[1]
+	eachMap := existingValues[stripCount(name)]
 	if eachMap != cty.NilVal {
+		if !eachMap.Type().IsObjectType() && !eachMap.Type().IsMapType() {
+			e.logger.WithFields(logrus.Fields{
+				"block": b.Label(),
+			}).Debugf("skipping unexpected cty value type '%s' for existing for_each context value", eachMap.GoString())
+
+			ob[*k] = b.Values()
+			return cty.ObjectVal(ob)
+		}
+
 		for e, v := range eachMap.AsValueMap() {
 			ob[e] = v
```

Comment:
I guess it's better to rename this `e` variable as it's the same as the struct now.

Reply:
good catch 👍